File: src/test/java/br/com/blz/testjava/domain/factory/WarehousesFactoryTest.java
package br.com.blz.testjava.domain.factory;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;

import java.util.List;

import org.junit.jupiter.api.Test;

import br.com.blz.testjava.domain.model.Warehouse;

class WarehousesFactoryTest {

    @Test
    void shouldCreateAZeroQuantityWarehouseListWithOneWarehouse() {
        List<Warehouse> warehouses = WarehousesFactory.zeroQuantityWarehouse();

        assertFalse(warehouses.isEmpty());
        assertEquals(1, warehouses.size());
        assertEquals(0, warehouses.stream().mapToInt(Warehouse::getQuantity).sum());
    }
}

File: src/test/kotlin/br/com/blz/testjava/application/factory/WarehousesFactoryTest.kt
package br.com.blz.testjava.application.factory

import br.com.blz.testjava.application.model.WarehouseModel
import br.com.blz.testjava.domain.entity.enums.WarehouseType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class WarehousesFactoryTest {

    @Test
    fun `It should create a Warehouses domain from a Warehouses model`() {
        val model = WarehouseModel("Acre", 12, WarehouseType.ECOMMERCE)
        val factory = WarehousesFactory()
        val domain = factory.create(model)

        assertEquals(domain.locality, "Acre")
        assertEquals(domain.quantity, 12)
        assertEquals(domain.type.name, "ECOMMERCE")
    }

}
